ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufDataTranslator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+
+package io.confluent.ksql.serde.protobuf;
+
+import io.confluent.ksql.serde.SchemaFullNameAppender;
+import io.confluent.ksql.serde.connect.ConnectDataTranslator;
+import io.confluent.ksql.serde.connect.DataTranslator;
+import io.confluent.ksql.util.DecimalUtil;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+
+public class ProtobufDataTranslator implements DataTranslator {
+  private final DataTranslator innerTranslator;
+  private final Schema ksqlSchema;
+  private final Schema protoCompatibleSchema;
+
+  ProtobufDataTranslator(final Schema schema, final String schemaFullName) {
+    this.ksqlSchema = Objects.requireNonNull(schema, "schema");
+
+    this.protoCompatibleSchema = SchemaFullNameAppender.appendSchemaFullName(
+        schema, schemaFullName
```

Comment:
Seems we only care about the schema name here, not sure if we need to call `appendSchemaFullName` which builds other changes.

Reply:
The `appendSchemaFullName` just adds the new name to the passed schema, right? It has other changes to replace `.` for `_`, which Avro and Protobuf don't support. I just found the bug in Protobuf https://github.com/confluentinc/ksql/issues/9005

Without this full name added to the schema, the INSERT fails too.